### PR TITLE
[Merged by Bors] - ET-3605 update rust toolchain 1.65.0

### PR DESCRIPTION
**Pull request by janpetschexain**
**Reference**

- [ET-3605]

**Summary**

the changes are grouped by each commit:
- update toolchain to `rustc 1.65.0`/`nightly-2022-11-03`, `just 1.8.0`, `cargo-sort 1.0.9`
- update rust deps and fix new lints, especially:
  - lindera: use released version instead of rev and fix breaking changes
  - criterion: fix broken benchmarks
  - chrono: fix deprecation warnings
- update github workflow deps
- release docker ci image `v6`

[ET-3605]: https://xainag.atlassian.net/browse/ET-3605?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine_core/Cargo.toml [hunk 1]
```diff
@@ -10,48 +10,41 @@ members = [
 [workspace.package]
 version = "0.1.0"
 edition = "2021"
-rust-version = "1.64.0"
+rust-version = "1.65.0"
 
 [workspace.dependencies]
-anyhow = "1.0.65"
-async-trait = "0.1.57"
+actix-web = { version = "4.2.1", default-features = false, features = ["compress-gzip", "compress-zstd"] }
```
Comment (acrrd) [inline discovery_engine_core/Cargo.toml:16]: actix is not a common dependency so it shouldn't be here.

Comment (acrrd) [reply discovery_engine_core/Cargo.toml:16]: Ah ok. didn't know it was used also in ` tooling`

**Review by acrrd — COMMENTED** [review]
My only problem is the changes from the commit hash to the tag. We are using the commit for security concerns and reproducibility,  I don't see a good reason to change it.

File: .github/actions/setup-job-docker/action.yml [hunk 2]
```diff
@@ -14,9 +14,9 @@ runs:
   using: "composite"
   steps:
     - name: Load .env
-      uses: xom9ikk/dotenv@d3ff95524814ceac377510f30f4af6296ea612c1 #v1.0.2
```
[the remark this answers is not in the document]
Comment (acrrd) [reply .github/actions/setup-job-docker/action.yml:17]: We should use the ref then :)

[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply .github/actions/setup-job-docker/action.yml:17]: btw dependabot now also automatically updates the version in the comment https://github.blog/changelog/2022-10-31-dependabot-now-updates-comments-in-github-actions-workflows-referencing-action-versions/

[the remark this answers is not in the document]
Comment (janpetschexain) [reply .github/actions/setup-job-docker/action.yml:17]: > btw dependabot now also automatically updates the version in the comment https://github.blog/changelog/2022-10-31-dependabot-now-updates-comments-in-github-actions-workflows-referencing-action-versions/

that's nice! makes it less annoying to update both the ref and the comment. do we plan to use dependabot again?

**Comment by janpetschexain** [comment]
> My only problem is the changes from the commit hash to the tag. We are using the commit for security concerns and reproducibility, I don't see a good reason to change it.

changed all actions to ref. just having the tag makes it easier to update for us, otherwise we have to update both ref and version comment. but i definitely agree that the ref is safer that the tag. maybe we can later ease the process by using dependabot again.

File: .github/actions/setup-job-docker/action.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (acrrd) [reply .github/actions/setup-job-docker/action.yml:17]: if now it is able to update the comment we could enable it.

**Review by acrrd — APPROVED** [review]

**Pull request closed without merging**